src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -290,37 +308,59 @@ CLASS ZCL_ABAPGIT_GIT_TRANSPORT IMPLEMENTATION.
     lo_client->set_headers( iv_url     = iv_url
                             iv_service = c_service-upload ).
 
-    LOOP AT lt_branches FROM 1 ASSIGNING <ls_branch>.
+    IF lt_commits IS NOT INITIAL.
+      MOVE-CORRESPONDING lt_commits TO lt_wanteds.
+    ELSE.
+      lt_wanteds = lt_branches.
+    ENDIF.
+
+    LOOP AT lt_wanteds FROM 1 ASSIGNING <ls_wanted>.
       IF sy-tabix = 1.
         lv_capa = 'side-band-64k no-progress multi_ack' ##NO_TEXT.
-        lv_line = 'want' && ` ` && <ls_branch>-sha1
+        lv_line = 'want' && ` ` && <ls_wanted>-sha1
           && ` ` && lv_capa && zif_abapgit_definitions=>c_newline. "#EC NOTEXT
       ELSE.
-        lv_line = 'want' && ` ` && <ls_branch>-sha1
+        lv_line = 'want' && ` ` && <ls_wanted>-sha1
           && zif_abapgit_definitions=>c_newline.            "#EC NOTEXT
       ENDIF.
       lv_buffer = lv_buffer && zcl_abapgit_git_utils=>pkt_string( lv_line ).
     ENDLOOP.
 
     IF iv_deepen = abap_true.
-      lv_buffer = lv_buffer && zcl_abapgit_git_utils=>pkt_string( 'deepen 1'
-        && zif_abapgit_definitions=>c_newline ).            "#EC NOTEXT
+      lv_buffer = lv_buffer && zcl_abapgit_git_utils=>pkt_string( |deepen { iv_deepen_level }| &&
+        |{ zif_abapgit_definitions=>c_newline }| ).         "#EC NOTEXT
     ENDIF.
 
     lv_buffer = lv_buffer
              && '0000'
              && '0009done' && zif_abapgit_definitions=>c_newline.
 
-    lv_xstring = lo_client->send_receive_close( zcl_abapgit_convert=>string_to_xstring_utf8( lv_buffer ) ).
+    TRY.
+        lv_xstring = lo_client->send_receive_close( zcl_abapgit_convert=>string_to_xstring_utf8( lv_buffer ) ).
 
-    parse( IMPORTING ev_pack = lv_pack
-           CHANGING cv_data = lv_xstring ).
+        parse( IMPORTING ev_pack = lv_pack
+               CHANGING  cv_data = lv_xstring ).
 
-    IF lv_pack IS INITIAL.
-      zcx_abapgit_exception=>raise( 'empty pack' ).
-    ENDIF.
+        IF lv_pack IS INITIAL.
+          zcx_abapgit_exception=>raise( 'Response could not be parsed - empty pack returned.' ).
+        ENDIF.
+
+        et_objects = zcl_abapgit_git_pack=>decode( lv_pack ).
+
+      CATCH zcx_abapgit_exception INTO lx_exception.
+        IF   io_repo IS BOUND
```

Comment:
this seems suspicious, git does not return an error that can be used?

Reply:
Obsolete - removed reference to repo completely.